api.ts

```diff
@@ -736,6 +746,19 @@ export interface SyncFormulaDef<
    */
   /** @hidden */
   maxUpdateBatchSize?: number;
+  /**
+   * The JavaScript function that implements this sync update if the table supports updates.
+   *
+   * This function takes in parameters, updated sync table objects, and a sync context, 
+   * and is responsible for pushing those updated objects to the external system then returning
+   * the new state of each object.
+   */
+  /** @hidden */
+   executeUpdate?(
+    params: ParamValues<ParamDefsT>,
+    updates: Array<SyncUpdate<K, L, SchemaT>>,
+    context: SyncExecutionContext,
```

Comment:
We may want a dedicated type for this context. The current type mostly applies here but it has `continuation` which is confusing, but at least it's optional.

Reply:
Added - did you have something like this in mind?